design/behavior.system.md

```diff
@@ -0,0 +1,95 @@
+# Behavior System
+
+The behavior system is responsible for provinding the abstracted user interaction model for MRE Apps and how the user can interact with
+the various components of an MRE App.  It is based on the concept of high level behaviors that are applied to actors to make the actor
+"behave" like that particular interaction behavior.
+
+## Architecture
+
+The system is comprised of several layers and components that make up the system.
+
+### SDK Components and Layers
+
+- Behavior
+    - Actions
+    - Action Handlers
+    - Action Data
+
+### Runtime Components and Layers
+
+- Behavior Context
+    - Action Handlers
+    - Action Data
+- Behavior Plugin Interface
+- Behavior Factory Plugin Interface
+
+## Network
+
+- SetBehavior Payload
+- PerformAction Payload
+
+## Sync layer considerations
+
+We sync behaviors from all clients in the server.  All filtering is handled in the runtime.
+
+## Client implementation
+
+Client runtime is resposible for getting the concrete host app implementation of the behaviors through the host app implementation
+of the `IBehaviorFactory`, and supplying the necesssary context for the host app to raise input events on.  The permission system is
+responsible for enabling or disabling the behavior system per user, or globally on the client if there is no user interacting with the
+app at all for that client.
+
+### Permission System Handling Per Case on a Client
+
+1.  Single User No User Interaction Permission Given
+
+Since there is no other users in the host app that have enabled user interaction, we do not add behaviors on this client when they are 
+set from the App.
+
+2. Single User With User Interaction Permission Given
+
+Client will add behaviors whenever they are set from the app and will send events when the user interacts with them inside the host app
+implementation.
+
+3. Multiple Users Where a Subset Have Given Permission
+
+Client will add behaviors to the actors as they are set from the app.  The runtime will filter events from the host app by user id, and
+send only the events to the server for users that have given permission for user interaction.  It will be up to the host app implementation
+to handle any targetability and cursor interactions with this actor by way of the `IMixedRealityExtensionApp` API that exposes the method
+`IsInteractableForUser(IUser user)`.
+
+4. Multiple Users Where All Users Have Given User Interaction Permission
+
+Client will add behaviors to all actors as thay are set from the app, and will do no filtering of any of the events.
+
+5. Multipl Users Where None Have Given User Interaction Permission
+
+Client will not add any behaviors to actors when they are set from the app, as there are no users interacting with the app on this client.
+
+## Example Usage
```

Comment:
>Example Usage [](start = 3, length = 13)

The first one doesn't have a behavior, the second one is new syntax (?), and the third one isn't right because it doesn't have a user argument.

Reply:
I stole this from the pen behavior that needs to be updated.  My bad.

---
In reply to: [448611299](https://github.com/microsoft/mixed-reality-extension-sdk/pull/649#discussion_r448611299) [](ancestors = 448611299)